test/minitest/test_minitest_benchmark.rb

```diff
@@ -118,6 +118,54 @@ def test_fit_power_noisy
     assert_fit :power, x, y, 0.96, 3.119e-5, 0.8959
   end
 
+  def test_sigma
+    bench = Minitest::Benchmark.new :blah
+    assert_equal bench.sigma([1,2,3]), 6
+    assert_equal bench.sigma([1,2,3]) { |n| n**2 }, 14
+  end
+
+  def test_fit_error
+    bench = Minitest::Benchmark.new :blah
+    assert_equal bench.fit_error([[1, 1], [2, 2], [3, 3]]) { |x| x }, 1.0
+    assert_in_delta bench.fit_error([[1, 1], [2, 2], [3, 4]]) { |x| x }, 0.785
+  end
+
+  def test_fit_linear_x2
+    bench = Minitest::Benchmark.new :blah
+    xs = [1, 10, 100, 1000]
+    _a, _b, rr = bench.fit_linear(xs, xs.map { |x| x**2 }), :<, 0.99
+    assert_operator rr, :<, 0.99
+  end
+
+  def test_fit_linear_x3
+    bench = Minitest::Benchmark.new :blah
+    xs = [1, 10, 100, 1000]
+    _a, _b, rr = bench.fit_linear(xs, xs.map { |x| x**3 })
+    assert_operator rr, :<, 0.99
+  end
+
+  def test_fit_constant_beyond_linear
+    bench = Minitest::Benchmark.new :blah
+    # This is actual data from a Minitest::Benchmark run (testing Tree#push)
+    # It sure looks steeper than linear slope 1.
+    # Look at y go from .05 to .26 as x goes from 1000 to 2000
+    xs = [10, 100, 1000, 2000]
+    ys = [0.000092, 0.000618, 0.050752, 0.258891]
+
+    # I don't know how to call #assert_performance_constant on static data
+    # i.e. without doing timing
+    # Here is what assert_performance_constant tests
+    _a, b, _rr = bench.fit_linear(xs, ys)
+    assert_in_delta 0, b, 0.01
```

Comment:
```
In[34]:= LinearModelFit[{{10, 9.2*10^-5}, {100, 0.000618}, {1000, 
    0.050752}, {2000, 0.258891}}, x, x]["RSquared"]

Out[34]= 0.907874
```

which is what minitest is outputting.

Reply:
Here's what it looks like:

![screen shot 2017-12-09 at 20 43 11](https://user-images.githubusercontent.com/9832/33801996-b26e43d6-dd21-11e7-9b33-538117e15ed6.png)

I'm not sure what this test is trying to say... it is indeed NOT linear and NOT constant... But the `xs` is very oddly picked and, if anything, too few are available to get a good fit that is free from runtime / machine noise.